src/main/java/tictactoe/Minimax.java

```diff
@@ -0,0 +1,81 @@
+package tictactoe;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Minimax {
+
+    public int chooseMove
+            (Board board, String currentPlayersMark, String opponentsMark) {
+
+        var copyBoard = board.copyBoard();
+        return findBestMove(copyBoard, 0, currentPlayersMark, opponentsMark);
+    }
+
+    private int findBestMove
+            (Board board, int depth, String currentPlayersMark, String opponentsMark) {
+        var movesAndScores = new HashMap<Integer, Integer>();
+        var availableSquares = board.availableSquares(currentPlayersMark, opponentsMark);
+
+        if (board.isComplete(currentPlayersMark, opponentsMark)) {
+            return scoreMove(board, depth, currentPlayersMark, opponentsMark);
+        }
+
+        for (String square : availableSquares) {
+            var squareNum = Integer.parseInt(square);
+            var copyBoard = board.copyBoard();
+
+            copyBoard.takeSquare(squareNum, currentPlayersMark);
+
+            movesAndScores.put(squareNum,
+                    -1 * findBestMove(copyBoard, depth + 1, opponentsMark, currentPlayersMark));
+        }
+        return evaluateMove(depth, movesAndScores);
+    }
+
+
+    public int maximisingPlayerBestMove(Map<Integer, Integer> scores) {
```

Comment:
Suppose this method is passed the following map:

```
move | score
-------------
3        | 2
4        | 4
5        | 10
9        | 7
```

What value do you expect to be assigned to `bestMove`? Is that the value which will actually be assigned given this code?

Reply:
I would expect 5 to be returned as the best score. However, it doesn't, because currently the value assigned to `bestScore` doesn't change when `score` is more than `bestScore`. I'll fix this, thanks for spotting it!